### PR TITLE
[FIX]: asynchronous context manager wrapper around openai.AsyncStream

### PR DESCRIPTION
**Pull request by MohanadAbugharbia**
### Overview:
Langchain OpenAi Async Stream breaks when openlit is initialised in a project

Breaking commit from langchain-openai [a682f0d](https://github.com/langchain-ai/langchain/commit/a682f0d12bdffa95db0ed424172db9815d76374d)

This PR wraps the output of the `openai.AsyncOpenAI.chat.completions.create` method of type `openai.AsyncStream`  with the `TracedAsyncStream` which is an asynchronous context manager and an asynchronous iterator that iterates on `openai.AsyncStream` and returns the results after some processing. 

The `TracedAsyncStream` replaces the asynchronous `stream_generator` function that used to iterate over the `openai.AsyncStream` 

Discussed in #391 

Error:
```bash
(.venv) ➜  openlit git:(main) ✗ python langchain_with_openai.py
Traceback (most recent call last):
  File "/Users/*********/programming/openlit/langchain_with_openai.py", line 28, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.11/3.11.9_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.9_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.9_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/*********/programming/openlit/langchain_with_openai.py", line 23, in main
    async for response in get_chat_response("What is the capital of France?"):
  File "/Users/*********/programming/openlit/langchain_with_openai.py", line 18, in get_chat_response
    async for response in completion:
  File "/Users/*********/programming/openlit/.venv/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 494, in astream
    raise e
  File "/Users/*********/programming/openlit/.venv/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 472, in astream
    async for chunk in self._astream(
  File "/Users/*********/programming/openlit/.venv/lib/python3.11/site-packages/langchain_openai/chat_models/base.py", line 2038, in _astream
    async for chunk in super()._astream(*args, **kwargs):
  File "/Users/*********/programming/openlit/.venv/lib/python3.11/site-packages/langchain_openai/chat_models/base.py", line 810, in _astream
    async with response:
TypeError: 'async_generator' object does not support the asynchronous context manager protocol
(.venv) ➜  openlit git:(main) ✗
```

### Visuals (If applicable):
Not applicable

### Checklist:
- [x] PR name follows conventional commit format: `[Feat]: ...` or `[Fix]: ....`
- [x] Added visuals for changes (If applicable)
- [x] Checked OpenLIT [contribution guidelines](https://github.com/openlit/openlit/blob/main/CONTRIBUTING.md)

## Summary by Sourcery

Bug Fixes:
- Fix the issue with Langchain OpenAi Async Stream by replacing the async_generator with TracedAsyncStream to support the asynchronous context manager protocol.

### CODE REVIEW

**Comment by MohanadAbugharbia** [comment]
@patcher9 should I update the version to `1.30.0` or `1.29.1`?

**Comment by amanagarwal042** [comment]
Mega work @MohanadAbugharbia, Just tested and it works!!!!!! Thank you so much for helping out with this. 


> @patcher9 should I update the version to 1.30.0 or 1.29.1?

You can update the version to `1.29.1`

PR looks good, With the version bump this should be ready to merge 🚀

**Review by amanagarwal042 — DISMISSED** [review]

**Comment by MohanadAbugharbia** [comment]
Alright, now I've fixed all the linting issues, let me know if I need to do something else.

File: sdk/python/pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "openlit"
-version = "1.29.0"
+version = "1.29.1"
```
Comment (amanagarwal042) [inline sdk/python/pyproject.toml]:
```suggestion
version = "1.29.2"
```
My bad, just released 1.29.1 so we can use `1.29.2` here

**Review by amanagarwal042 — DISMISSED** [review]

**Review by amanagarwal042 — APPROVED** [review]

**Comment by amanagarwal042** [comment]
🥳 
Also you can join our Slack community if you wanna discuss implementing this structure for other Async Wrappers aswell
Link - https://join.slack.com/t/openlit/shared_invite/zt-2etnfttwg-TjP_7BZXfYg84oAukY8QRQ

**Pull request merged**